src/test/groovy/graphql/execution/ExecutionConstraintsTest.groovy

```diff
@@ -0,0 +1,276 @@
+            hero: [name   : 'R2-D2',
+                   friends: [
+                           [
+                                   name     :
+                                           'Luke Skywalker',
+                                   appearsIn: ['NEWHOPE', 'EMPIRE', 'JEDI'],
+                                   friends  : [
+                                           [
+                                                   name: null,
+                                           ],
+                                           [
+                                                   name: null,
+                                           ],
+                                           [
+                                                   name: null,
+                                           ],
+                                           [
+                                                   name: null,
+                                           ],
+                                   ]
+                           ],
+                           [
+                                   name     : 'Han Solo',
+                                   appearsIn: ['NEWHOPE', 'EMPIRE', 'JEDI'],
+                                   friends  : [
+                                           [
+                                                   name: null,
+                                           ],
+                                           [
+                                                   name: null,
+                                           ],
+                                           [
+                                                   name: null,
+                                           ],
+                                   ]
+                           ],
+                           [
+                                   name     : 'Leia Organa',
+                                   appearsIn: ['NEWHOPE', 'EMPIRE', 'JEDI'],
+                                   friends  : [
+                                           [
+                                                   name: null,
+                                           ],
+                                           [
+                                                   name: null,
+                                           ],
+                                           [
+                                                   name: null,
+                                           ],
+                                           [
+                                                   name: null,
+                                           ],
+                                   ]
+                           ],
+                   ]
+            ]
+    ]
+
+
+    def 'Shows full data when the max depth is not exceeded'() {
+        given:
+
+        when:
+        def result = GraphQL
+                .newGraphQL(StarWarsSchema.starWarsSchema)
+                .maxQueryDepth(99)
```

Comment:
Just to test that is not "unlimited" but rather not exceeded during the depth.  WQe have a case when its < 0 then its unlimited